### PR TITLE
[bitnami/postgresql] Use plain connections from metrics sidecar

### LINKED ISSUE #11822: [bitnami/postgresql] metrics not work if tls.enabled=true  and tls.certCAFilename not empty
### Name and Version

bitnami/postgres:11.7.1

### What steps will reproduce the bug?

values set:
``` yaml
auth:
    username: xxx
    password: xxx
    database: postgres

tls:
    enabled: true
    certificatesSecret: db-certificates-tls-secret
    certFilename: serverxxx.crt
    certKeyFilename: serverxxx.key
    certCAFilename: caxxx.crt
```
After installing and getting metrics will get the following error:
```
ts=2022-08-18T05:19:59.443Z caller=postgres_exporter.go:731 level=error err="Error opening connection to database (host=127.0.0.1%20port=5432%20user=xxx%20sslmode=require%20sslcert=/opt/bitnami/postgresql/certs/server.crt%20sslkey=/opt/bitnami/postgresql/certs/server.key): pq: database \"xxx\" does not exist"
```
Seems caused by no database name same as the custom username, and we can not specify the database for env `DATA_SOURCE_NAME`

### What do you see instead?

error logs:
```
ts=2022-08-18T05:19:59.443Z caller=postgres_exporter.go:731 level=error err="Error opening connection to database (host=127.0.0.1%20port=5432%20user=xxx%20sslmode=require%20sslcert=/opt/bitnami/postgresql/certs/server.crt%20sslkey=/opt/bitnami/postgresql/certs/server.key): pq: database \"xxx\" does not exist"
```
### Additional information

If can specifies the database for env `DATA_SOURCE_NAME` in values, this issue can be resolved

### PR DESCRIPTION
**Pull request by fmulero**
Signed-off-by: Fran Mulero <fmulero@vmware.com>

### Description of the change

Use always plain connections from metrics sidecar.

### Benefits

Current configuration causes errors retrieving because we are using the certificate configured in the server side as client certificate in the metrics exporter, and they are not the same. A client certificate should have as `cn` the name of the user to be authenticated. 

### Possible drawbacks

In a deployment with the following configuration, the upgrade will fail (the metrics exporter could not retrieve data):
* Custom `pg_hba.conf` forcing ssl everywhere
* The CN for the server certificate and the CN for a user with enough privileges are the same (this seems weird but a lot of people use `postgres` everywhere)

### Applicable issues

  - fixes #11822

### Additional information

Without these changes, metrics with the autogenerated certs seem to be working, but TLS is disabled because no `sslcert` or `sslkey` is provided and the log shows:

```
ts=2022-09-29T13:53:17.116Z caller=tls_config.go:195 level=info msg="TLS is disabled." http2=false
``` 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql/templates/primary/statefulset.yaml [hunk 1]
```diff
@@ -488,14 +488,8 @@ spec:
...
             {{- $database := required "In order to enable metrics you need to specify a database (.Values.auth.database or .Values.global.postgresql.auth.database)" (include "postgresql.database" .) }}
-            {{- $sslmode := ternary "require" "disable" .Values.tls.enabled }}
-            {{- if and .Values.tls.enabled .Values.tls.certCAFilename }}
-            - name: DATA_SOURCE_NAME
-              value: {{ printf "host=127.0.0.1 port=%d user=%s sslmode=%s sslcert=%s sslkey=%s" (int (include "postgresql.service.port" .)) (default "postgres" $customUser) $sslmode (include "postgresql.tlsCert" .) (include "postgresql.tlsCertKey" .) }}
-            {{- else }}
             - name: DATA_SOURCE_URI
-              value: {{ printf "127.0.0.1:%d/%s?sslmode=%s" (int (include "postgresql.service.port" .)) $database $sslmode }}
-            {{- end }}
+              value: {{ printf "127.0.0.1:%d/%s?sslmode=require" (int (include "postgresql.service.port" .)) $database }}
```
Comment (migruiz4) [inline bitnami/postgresql/templates/primary/statefulset.yaml:492]: Shouldn't this be `sslmode=disable` instead of `require`?

**Review by migruiz4 — COMMENTED** [review]
Could you please take a look at my comment? 
I'm not sure why primary is set with `ssl=require` and the read replica is set with `ssl=disable`.

File: bitnami/postgresql/templates/primary/statefulset.yaml (hunk 1 above)
Comment (fmulero) [reply bitnami/postgresql/templates/primary/statefulset.yaml:492]: My fault! I was testing the same configuration that we had when `tls.autoGenerated` is enabled and I pushed it.

File: bitnami/postgresql/Chart.yaml [hunk 2]
```diff
@@ -28,4 +28,4 @@ name: postgresql
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/postgresql
   - https://www.postgresql.org/
-version: 11.9.3
+version: 11.9.4
```
Comment (migruiz4) [inline bitnami/postgresql/Chart.yaml:31]: Please rebase your branch and set a new chart version: https://github.com/bitnami/charts/blob/master/bitnami/postgresql/Chart.yaml

**Review by migruiz4 — CHANGES_REQUESTED** [review]

File: bitnami/postgresql/Chart.yaml (hunk 2 above)
Comment (migruiz4) [reply bitnami/postgresql/Chart.yaml:31]: The latest version is now 11.9.5

**Review by migruiz4 — APPROVED** [review]

**Pull request merged**